### PR TITLE
maryam

### PR DESCRIPTION
**Pull request by maryam-alraddadi**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        private static Singleton _singleton;
+
+        public Singleton()
+        {
+        }
+
+         public static Singleton getInstance()
+         {
+             return _singleton ?? new Singleton();
```
Comment (SN-SAFCSP) [inline Program.cs:17]: here I think you forgot the equal sign `=`
it should be like this` return _singleton ??= new Singleton();`
that is why your code shows `False` and two different instances.

**Comment by sarahAlqahtani** [comment]
**Great Job Maryam!**
Good Job implementing the Singleton pattern :clap::clap:, but you didn't make it thread safe.

Here are some resources on how to make your Singleton Pattern thread safe
[Thread-Safe Example](https://csharpindepth.com/Articles/Singleton)

> don't forget to check comments on your code.

File: Program.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+using System.Runtime.CompilerServices;
+
+namespace Singleton
+{
+
+    class Singleton
+    {
+        private static Singleton _singleton;
+
+        public Singleton()
```
Comment (SN-SAFCSP) [inline Program.cs:11]: the Constructor should be `private`

**Comment by SN-SAFCSP** [comment]
**Great Job Maryam!**
Good Job implementing the Singleton pattern :clap::clap:, but you didn't make it thread safe.

Here are some resources on how to make your Singleton Pattern thread safe
[Thread-Safe Example](https://csharpindepth.com/Articles/Singleton)

> don't forget to check comments on your code.

**Pull request closed without merging**